src/backend/renderer/element/surface.rs

```diff
@@ -336,31 +345,52 @@ impl<R: Renderer + ImportAll> WaylandSurfaceRenderElement<R> {
         kind: Kind,
     ) -> Result<Self, <R as Renderer>::Error>
     where
-        <R as Renderer>::TextureId: 'static,
+        <R as Renderer>::TextureId: Clone + 'static,
     {
         let id = Id::from_wayland_resource(surface);
         crate::backend::renderer::utils::import_surface(renderer, states)?;
 
+        let data_ref = states
+            .data_map
+            .get::<RendererSurfaceStateUserData>()
+            .map(|data| data.borrow());
+        let mut fallback_data = None;
+
+        let data = data_ref.as_deref().unwrap_or_else(|| {
+            fallback_data = Some(RendererSurfaceState::default());
+            fallback_data.as_ref().unwrap()
+        });
```

Comment:
I just realized that we could get rid of this here and also most `Option` handling if we just return `Result<Option<Self>, <R as Renderer>::Error>` in case the surface is not mapped. `render_elements_from_surface_tree` already skips all surfaces that have no `view` anyway. This could result in cleaner code in the `Element` implemetation.

Reply:
Not a bad idea. I'll try that asap

Reply:
thanks!